src/zino/api/legacy.py

```diff
@@ -29,14 +33,21 @@ def requires_authentication(func: Callable) -> Callable:
 class Zino1BaseServerProtocol(asyncio.Protocol):
     """Base implementation of the Zino 1 protocol, with a basic command dispatcher for subclasses to utilize."""
 
-    def __init__(self, state: Optional[ZinoState] = None, secrets_file: Optional[Union[Path, str]] = "secrets"):
+    def __init__(
+        self,
+        server: Optional["ZinoServer"] = None,
+        state: Optional[ZinoState] = None,
+        secrets_file: Optional[Union[Path, str]] = "secrets",
+    ):
         """Initializes a protocol instance.
 
         :param state: An optional reference to a running Zino state that this server should be based on.  If omitted,
                       this protocol will create and work on an empty state object.
         :param secrets_file: An optional alternative path to the file containing users and their secrets.
         """
+        self.server = server
```

Comment:
Add ":param server: An optional instance of ZinoServer" above.

Reply:
fixed